pkg/ipam/types/types.go

```diff
@@ -114,6 +114,15 @@ type IPAMStatus struct {
 	//
 	// +optional
 	OperatorStatus OperatorStatus `json:"operator-status,omitempty"`
+
+	// ReleaseIPs tracks the state for every IP considered for release.
+	// value can be one of the following string :
+	// * marked-for-release : Set by operator as possible candidate for IP
+	// * ready-for-release  : Acknowledged as safe to release by agent
+	// * do-not-release     : IP already in use / not owned by the node. Set by agent
+	//
+	// +optional
+	ReleaseIPs map[string]string `json:"release-ips,omitempty"`
```

Comment:
nit: You could introduce a newtype with K8s validation requirements here. For example:

```go
// +kubebuilder:validation:Enum=marked-for-release;ready-for-release;do-not-release
type IPReleaseStatus string
```

And then change this type to:

```suggestion
	ReleaseIPs map[string]IPReleaseStatus `json:"release-ips,omitempty"`
```

Reply:
Added the new type and validations. Had to update some tests to convert from `IPReleaseStatus ` to string